.github/workflows/lint.yml

```diff
@@ -0,0 +1,28 @@
+name: Lint code
```

Comment:
I suspect it would have been easier to leave this more or less as it was via `pre-commit` and just had a second `.non-pre-commit-ci.yaml` file and used `pre-commit run --config` -- in fact I suspect that would have been a two line change to the existing setup (just adding a second `uses: pre-commit/action` via [custom invocation](https://github.com/pre-commit/action#using-this-action-with-custom-invocations) to include the extra config file.

The advantage again would have been that someone can still run this without needing to read this workflow and figure out how to reproduce it to run the linter.

But it's not a huge deal I suppose, so if you've done the work already here and it works (which it seems to) I'm ok with merging -- though if you do merge, please rename the file and workflow title to something quite a bit more "Java linters"-specific, and yeah can we put `pmd-ruleset.xml` somewhere way less conspicuous, e.g. `.java-implementations-pmd-ruleset.xml` or `implementations/java-implementations-pmd-ruleset.xml` or something?

Reply:
I mean, the changes would have been way narrower, but I decided to give PMD a try instead of checkstyle